drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -44,46 +44,47 @@ end
 --- and supports the OnOff cluster. This is done to bypass the
 --- BRIDGED_NODE_DEVICE_TYPE on bridged devices
 local function find_default_endpoint(device, component)
-  local res = device.MATTER_DEFAULT_ENDPOINT
   local eps = device:get_endpoints(clusters.OnOff.ID)
   table.sort(eps)
   for _, v in ipairs(eps) do
     if v ~= 0 then --0 is the matter RootNode endpoint
-      res = v
-      break
+      return v
     end
   end
   device.log.warn(string.format("Did not find default endpoint, will use endpoint %d instead", device.MATTER_DEFAULT_ENDPOINT))
-  return res
+  return device.MATTER_DEFAULT_ENDPOINT
 end
 
-local function initialize_switch(device)
+local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
   table.sort(switch_eps)
-  local component_map = {}
 
-  -- For switch devices, the profile components follow the naming convention "switch%d",
-  -- with the exception of "main" being the first component. Each component will then map
-  -- to the next lowest endpoint that hasn't been mapped yet.
-  -- Additionally, since we do not support bindings at the moment, we only want to count
-  -- On/Off clusters that have been implemented as server. This can be removed when we have
+  -- Since we do not support bindings at the moment, we only want to count On/Off
+  -- clusters that have been implemented as server. This can be removed when we have
   -- support for bindings.
   local num_server_eps = 0
+  local main_endpoint = find_default_endpoint(device)
   for _, ep in ipairs(switch_eps) do
     if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1;
-      if num_server_eps == 1 then
-        component_map["main"] = ep
-      else
-        component_map[string.format("switch%d", num_server_eps)] = ep
+      num_server_eps = num_server_eps + 1
+      if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
+        local name = string.format("%s %d", device.label, num_server_eps)
+        driver:try_create_device(
+          {
+            type = "EDGE_CHILD",
+            label = name,
+            profile = "switch-binary",
+            parent_device_id = device.id,
+            parent_assigned_child_key = string.format("%02d", ep),
```

Comment:
I have seen elsewhere in our drivers that the `parent_assigned_child_key` is converted to hex with something like `parent_assigned_child_key = string.format("%02X", i)`, however this will not work here due to an issue in the lua libs where the `MatterChildDevice:get_endpoint()` function does not convert the child key as base 16. In this case, any child key for any endpoint above 10 would not be converted correctly back to a number.

This can be address with a simple change in the lua libs: https://github.ecodesamsung.com/iot-hub/scripting-engine/pull/1608, however, to get this working without the lua libs change I have just used base 10 for the parent key for now.

Reply:
I dont think we should be aiming to switch the way we store these values, since it would be an api breaking change. ~Limiting the endpoint id to two digits is not spec compliant since an endpoint id is a u16 which can be > 99~ Looks like `"%02X` isn't actually limiting, it is just 0 padding, but I still think my suggested change is valid, since the 0 padding is not typically done for base 10 numbers.
```suggestion
            parent_assigned_child_key = string.format("%d", ep),
```
Also, I dont think we should link to private repos here.

Reply:
Done!